sklearn/impute/_base.py

```diff
@@ -628,7 +645,15 @@ def _fit(self, X, y=None):
             The imputer mask of the original data.
 
         """
-        X = self._validate_input(X)
+        self._precomputed = False
+        if sparse.issparse(X) or isinstance(X, np.ndarray):
+            if X.dtype == 'bool' and self.missing_values:
```

Comment:
I should miss it but what is the reason for adding `self.missing_values` in the condition?

Reply:
In the beginning, we were trying not to use the `precomputed` flag. So, to make sure that a `mask` is being passed we were checking whether `self.missing_values==True` or not. 
But what I think the reason should be that when we set `precomputed=True`, we no longer want to use `validate_input` "again" as the `mask` has already been computed on "validated" input, but if a user (there also exists tests) that pass `boolean` array, so, if we just check the `dtype` and set `precomputed=true`, this would "bypass" the validation, and hence, lead to failure of tests and also is a source of error.